sklearn/decomposition/pca.py

```diff
@@ -216,6 +216,61 @@ def __init__(self, n_components=None, copy=True, whiten=False):
         self.copy = copy
         self.whiten = whiten
 
+    def get_feature_names(self, input_features=None, full_output=False):
+        """
+        Return dominant feature names for each component
+
+        Parameters
+        ----------
+        input_features : list of string, shape (n_features), optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        full_output : boolean, default False
+            When it is "True", return the principal components as the
+            combination of the input features. If "False", will be just
+            the component names
+
+        Returns
+        -------
+        output_feature_names : list of string, shape (n_components)
+            When full_output is "True", it is represented by the contribution
+            of input features like "a1 * x0 + a2 * x1", "b1 * x0 + b2 * x1".
+            If full_output is "False", it just represents the component names
+            like "pc0", "pc1"
+        """
+        check_is_fitted(self, 'components_')
+
+        n_features = self.components_.shape[1]
+        components = self.components_
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(n_features)]
+        else:
+            if len(input_features) != n_features:
+                raise ValueError("Length of input_features is {0} but it must"
+                    "equal number of features when fitted: {1}.".format
+                    (len(input_features), self.n_features))
+
+        def get_contr(component):
```

Comment:
I should definitely use formatting here, but then it will look like 
``` ['0.838 * x0 + 0.544 * x1', '0.544 * x0 -0.838 * x1'] ``` with no space between the negative sign and digit. Please bear my naivety in asking doubts about a trivial issue.